walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +272,155 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void noTimeSlotsWithOptionalAtendee() {
+    // Optinal atendee has an all-day event. 
+    // We should see the two options that fit the mandatory attendees, ignoring the optional atendee,
+    // because there is no slots available for both mandatory and optional attendees.
+    //
+    // Events of mandetory attendees  :        |--A--|     |--B--|
+    // Events of optional attendee    :  |-------------C---------------|
+    // Day                            :  |-----------------------------|
+    // Options to return              :  |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.WHOLE_DAY, Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+        request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+
+  @Test
+  public void availableTimeSlotsWithOptionalAtendee() {
+    // Optional atendee has an event so that one time slot that fits the mandatory attendees is not suitable for him,
+    // and two time slot fits bot mandatory and optional attendees. 
+    // Only those two time slots should be returned.
+    //
+    // Events of mandetory attendees  :        |--A--|     |--B--|
+    // Events of optional attendee    :              |--C--|
+    // Day                            :  |-----------------------------|
+    // Options to return              :  |--1--|                 |--2--|
+
+    Collection<Event> events = Arrays.asList(
```

Comment:
Adding this comment once (yet applies for all tests). The test structure should be:
1. Arrange
2. Act
3. Assert

Each part separated with a new line from the other. Notice here your structure should be:
// Arrange
event = ...
request = ...

// Act
Collection<TimeRange> output = query.query(events, request);

// Assert
    Collection<TimeRange> expected =
        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
    Assert.assertEquals(expected, actual);
or even avoid the extra param and do:
Assert.assertEquals(output, Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false), TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true)));

Reply:
Now the structure is much clearer to me, thanks!